GitUI/CommandsDialogs/SettingsDialog/Pages/ScriptsSettingsPage.cs

```diff
@@ -48,7 +48,8 @@ public partial class ScriptsSettingsPage : SettingsPageWithHeader
 {sAuthorDate}
 {sCommitDate}
 
-Current Branch:
+On currently checked out revision:
+{cHead} (checked out branch name or checked out commit hash)
```

Comment:
for HEAD I only expect the sha, never branchname
Affects the tests too if that is used

Reply:
Agree

Reply:
>I would prefer to change HEAD - expected cBranch to contain the branch name

That's the main problem of the existing variables that I try to achieve with this new variable. None return the "thing" checked out even if "current" means current checked out revision. They act on the revision grid and so {cBranch} open a branch selector and let the user select the branch he wants. Indeed if there is only one branch (and no remote either --so maybe you were more talking about `{cLocalBranch}`--), the selector is not open and the user is lucky.

So I would like a variable that returns the branch checked out. I thought I had a good idea to reproduce **exactly** the same behavior than `.git/HEAD` that contains the branch name or the commit hash (because HEAD is not a hash but a pointer on the currently check out "object" i.e. local ref or hash!) **and so giving it the exact same name**.

> for HEAD I only expect the sha, never branchname

For that, we already have {cHash}

But if you find it misleading, I'm happy to give a clearer (and maybe less git oriented, more beginner friendly) name. What do you think about `{checkedOutBranch}`? But in this case, we will loose the benefit to get the commit hash if on detached head. I don't know if it is important (because the value will be empty and will probably break some usages)...

Reply:
> `cBranchOrHead`?

I'm really not convinced by this proposition...
I don't think it stand out enough compare to {cLocalBranch} or {cBranch} where the logic is to let the user select a branch among all the refs (heads? 😉 ) available on the current revision.
And partly also because, for me HEAD is most of the time not a hash...

>for HEAD I only expect the sha, never branchname

So I think Head is perfect notion/word for that so I can go with {cHeadBranchOrHash} and maybe you should change your perception of HEAD because the [git book](https://git-scm.com/book/en/v2/Git-Internals-Git-References#ref_the_ref) is pretty clear on that:

>**Usually the HEAD file is a symbolic reference to the branch you’re currently on**. By symbolic reference, we mean that unlike a normal reference, it contains a pointer to another reference.

>**However in some ___RARE___ cases the HEAD file may contain the SHA-1 value of a Git object**. This happens when you checkout a tag, commit, or remote branch, which puts your repository in ["detached HEAD"](https://git-scm.com/docs/git-checkout#_detached_head) state.

Another proposition is maybe to use `{HEAD}` to be clear that the value will have the same behavior than git HEAD.

Reply:
No no no, I will not change!

OK, go with just HEAD then.